zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterManagedProcess.java

```diff
@@ -96,18 +116,66 @@ public int getPort() {
   @Override
   public void start(String userName, Boolean isUserImpersonate) {
     // start server process
+    final String callbackHost;
+    final int callbackPort;
     try {
       port = RemoteInterpreterUtils.findRandomAvailablePortOnAllLocalInterfaces();
       logger.info("Choose port {} for RemoteInterpreterProcess", port);
+      callbackHost = RemoteInterpreterUtils.findAvailableHostname();
+      callbackPort = RemoteInterpreterUtils.findRandomAvailablePortOnAllLocalInterfaces();
```

Comment:
I think it's not very uncommon that system administrator setup firewall with ports whitelisted.
Once Interpreter is run in remote machines, this randomly assigned port may make firewall setup difficult.

Any thoughts on this?

Reply:
Make sense, will update the PR.